src/plugins/charts/public/static/color_maps/color_maps.ts

```diff
@@ -54,3128 +62,87 @@ export interface ColorMap {
   [key: string]: RawColorSchema;
 }
 
+const COLOR_MAP_LENGTH = 512;
+
+function convertColorPaletteToColorMap(colorPalette: string[]): RawColorSchema['value'] {
+  const colorMap: RawColorSchema['value'] = [];
+
+  for (let i = 0; i < colorPalette.length; i++) {
+    const color = colorPalette[i];
+    const regex = /#([0-9a-zA-Z]{2})([0-9a-zA-Z]{2})([0-9a-zA-Z]{2})/;
+
+    // @ts-ignore
+    const [, rawRed, rawGreen, rawBlue] = regex.exec(color);
+    const [red, green, blue] = [
+      parseInt(rawRed, 16) / 255,
+      parseInt(rawGreen, 16) / 255,
+      parseInt(rawBlue, 16) / 255,
+    ];
+
+    colorMap[i] = [i / (colorPalette.length - 1), [red, green, blue]];
```

Comment:
Reviewers - `colorPalette.length - 1` is like that (with the `- 1`) to keep the range `[0, 1]` inclusive, to keep parity with the previous implementation